compiler/src/dotty/tools/dotc/transform/TreeChecker.scala

```diff
@@ -495,6 +495,18 @@ object TreeChecker {
       assert(tree.qual.typeOpt.isInstanceOf[ThisType], i"expect prefix of Super to be This, actual = ${tree.qual}")
       super.typedSuper(tree, pt)
 
+    override def typedNew(tree: untpd.New, pt: Type)(using Context): Tree =
+      val tree1 = super.typedNew(tree, pt).asInstanceOf[tpd.New]
+      val sym = tree1.tpe.typeSymbol
+      if !(ctx.phase <= postTyperPhase) then
```

Comment:
this would be clearer as:
```suggestion
      if ctx.phase > postTyperPhase then
```
but could you add a comment explaining why this check is needed?